### PR TITLE
Port "ensureToken" over to PTerm

### PR DESCRIPTION
**Pull request by DavidS-ovm**
This replaces most of https://github.com/overmindtech/cli/blob/main/cmd/root.go#L250-L377 (which is still used elsewhere for now).

Results:

2024-08-13_18MS+0200_1282x568

It's not fully 1/1 identical to the bubbletea behaviour, but it is SO. MUCH. LESS. CODE.

2024-08-14_16MS+0200_1057x139

### CODE REVIEW

**Review by dylanratcliffe — COMMENTED** [review]
That's way easier to understand from my perspective

**Comment by DavidS-ovm** [comment]
This will still require more work and fine-tuning before merging, but it already `overmind terraform apply`d successfully.

**Comment by DavidS-ovm** [comment]
This is now ready for final reviews.

File: cmd/theme.go [hunk 1]
```diff
@@ -376,33 +369,41 @@ func wrap(s string, width, indent int) string {
...
-		checkMark = "OK"
+		return "OK"
 	}
-	return lipgloss.NewStyle().Foreground(ColorPalette.BgSuccess).Render(checkMark)
+	return "✔︎"
 }
 
-func RenderUnknown() string {
-	checkMark := "?"
+func RenderOk() string {
```
Comment (tphoney) [inline cmd/theme.go:379]: this doesnt seem to be used anywhere

Comment (tphoney) [reply cmd/theme.go:379]: same with `func wrap(`

Comment (tphoney) [reply cmd/theme.go:379]: so i found this https://go.dev/blog/deadcode

which then gives more cleanup 

```vscode ➜ /workspace/cli (e469f1d) $ deadcode .
cmd/auth_client.go:16:6: unreachable func: AuthenticatedApiKeyClient
cmd/changes_submit_plan.go:52:31: unreachable func: plannedChangeGroups.NumUnsupportedChanges
cmd/changes_submit_plan.go:62:31: unreachable func: plannedChangeGroups.NumSupportedChanges
cmd/changes_submit_plan.go:72:31: unreachable func: plannedChangeGroups.MappedItemDiffs
cmd/changes_submit_plan.go:87:31: unreachable func: plannedChangeGroups.Add
cmd/logging.go:38:20: unreachable func: TextStyle.TextStyle
cmd/logging.go:52:16: unreachable func: Color.Color
cmd/root.go:58:28: unreachable func: OvermindInstance.String
cmd/theme.go:323:6: unreachable func: styleH2
cmd/theme.go:364:6: unreachable func: wrap
cmd/theme.go:379:6: unreachable func: RenderOk
cmd/theme.go:389:6: unreachable func: RenderUnknown
cmd/theme.go:399:6: unreachable func: RenderErr
tracing/main.go:197:6: unreachable func: LogRecoverToReturn
tracing/main.go:209:6: unreachable func: LogRecoverToExit
tracing/main.go:224:6: unreachable func: handleError
vscode ➜ /workspace/cli (e469f1d) $ 
```

File: cmd/terraform_apply.go [hunk 2]
```diff
@@ -129,344 +86,84 @@ func NewTfApplyModel(args []string, parent *cmdModel, width int) tea.Model {
...
 		changeUuid, err := getChangeUuid(ctx, oi, sdp.ChangeStatus_CHANGE_STATUS_DEFINING, ticketLink, true)
 		if err != nil {
-			m.startingChange <- fatalError{err: fmt.Errorf("failed to identify change: %w", err)}
-			return nil
+			return uuid.Nil, fmt.Errorf("failed to identify change: %w", err)
 		}
 
-		m.startingChange <- changeIdentifiedMsg{uuid: changeUuid}
-		m.startingChange <- m.startingChangeSnapshot.StartMsg()
+		// m.startingChange <- m.startingChangeSnapshot.StartMsg()
```
[the remark this answers is not in the document]
Comment (DavidS-ovm) [reply cmd/terraform_apply.go:144]: no

File: cmd/terraform_apply.go [hunk 3]
```diff
@@ -129,344 +86,84 @@ func NewTfApplyModel(args []string, parent *cmdModel, width int) tea.Model {
...
+		startingChangeSnapshotSpinner, _ := pterm.DefaultSpinner.WithWriter(multi.NewWriter()).Start("Starting Change")
```
[the remark this answers is not in the document]
Comment (DavidS-ovm) [reply cmd/terraform_apply.go:145]: Yes, a lot of functions in pterm have a error return signature, but never actually return an error, and all their example suggests to ignore the error.

**Comment by DavidS-ovm** [comment]
rebased on top of the latest go.mod changes

**Comment by dylanratcliffe** [comment]
Screenshot 2024-09-20 at 17 57 33

Dat ratio

File: cmd/explore.go [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+	log "github.com/sirupsen/logrus"
+	"github.com/sourcegraph/conc/pool"
+	"github.com/spf13/cobra"
+	"golang.org/x/oauth2"
+)
+
+// exploreCmd represents the explore command
+var exploreCmd = &cobra.Command{
+	Use:    "explore",
+	Short:  "Run local sources for using in the Explore page",
```
Comment (dylanratcliffe) [inline cmd/explore.go:25]: I don't think it should block the merge but I'd like @jameslaneovermind to look at the descriptions here and jazz them up a bit

**Review by dylanratcliffe — APPROVED** [review]

**Pull request merged**